client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -62,91 +65,59 @@ function getGeneralTabUrl( slug ) {
-		// to the server
 		dirtyFields: new Set(),
 	};
 
-	componentDidMount() {
-		this.refreshCustomTitles();
-	}
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		const { selectedSite: prevSite, isFetchingSite, translate } = this.props;
-		const { selectedSite: nextSite } = nextProps;
-		const { dirtyFields } = this.state;
+	static getDerivedStateFromProps( props, state ) {
+		const { dirtyFields } = state;
+		const nextState = {};
 
-		// save success
-		if ( this.state.isSubmittingForm && nextProps.isSaveSuccess ) {
-			this.props.markSaved();
-			this.props.requestSiteSettings( nextProps.siteId );
-			this.refreshCustomTitles();
-			this.setState( { isSubmittingForm: false } );
+		if (
+			! dirtyFields.has( 'seoTitleFormats' ) &&
+			! isEqual( props.storedTitleFormats, state.seoTitleFormats )
+		) {
+			nextState.seoTitleFormats = props.storedTitleFormats;
 		}
 
-		// save error
-		if ( this.state.isSubmittingForm && nextProps.saveError ) {
-			this.setState( { isSubmittingForm: false } );
-			this.props.errorNotice(
-				translate( 'There was a problem saving your changes. Please, try again.' ),
-				{
-					id: 'seo-settings-form-error',
-				}
-			);
+		if (
+			! dirtyFields.has( 'frontPageMetaDescription' ) &&
+			! isEqual(
+				props.selectedSite.options?.advanced_seo_front_page_description,
+				state.frontPageMetaDescription
+			)
+		) {
+			nextState.frontPageMetaDescription =
+				props.selectedSite.options?.advanced_seo_front_page_description;
 		}
 
-		// if we are changing sites, everything goes
-		if ( prevSite.ID !== nextSite.ID ) {
-			return this.setState(
-				{
-					...stateForSite( nextSite ),
-					seoTitleFormats: nextProps.storedTitleFormats,
-					dirtyFields: new Set(),
-				},
-				this.refreshCustomTitles
-			);
+		if ( Object.keys( nextState ).length > 0 ) {
+			return nextState;
 		}
 
-		let nextState = {
-			...stateForSite( nextProps.selectedSite ),
-			seoTitleFormats: nextProps.storedTitleFormats,
-		};
-
-		if ( ! isFetchingSite ) {
-			const nextDirtyFields = new Set( dirtyFields );
-			nextDirtyFields.delete( 'seoTitleFormats' );
+		return null;
+	}
 
-			nextState = {
-				...nextState,
-				seoTitleFormats: nextProps.storedTitleFormats,
-				dirtyFields: nextDirtyFields,
-			};
-		}
+	componentDidMount() {
+		this.refreshCustomTitles();
+		this._mounted = true;
+	}
 
-		if ( dirtyFields.has( 'seoTitleFormats' ) ) {
-			nextState = omit( nextState, [ 'seoTitleFormats' ] );
-		}
+	componentWillUnmount() {
+		this._mounted = false;
+	}
 
-		// Don't update state for fields the user has edited
-		nextState = omit( nextState, Array.from( dirtyFields ) );
+	handleSubmitSuccess() {
+		this.props.markSaved();
+		this.props.requestSiteSettings( this.props.siteId );
+		this.refreshCustomTitles();
 
-		this.setState( nextState );
+		if ( this._mounted ) {
```

Comment:
Hm, that doesn't seem to be true at least for class components (haven't checked functional ones). I tried to replicate the behavior in this [sandbox](https://codesandbox.io/s/exciting-ramanujan-6yvvr) and if you hit _Hide_ before the async "task" is done you still get an error in the console. I may be misunderstanding though. Let me know what you think.

Reply:
Yes, that's expected because you haven't added assigned that ref to an element.

If you assign `ref={ this.mounted }` to an element, React will set that ref to `null` when the component unmounts. Let me know if that makes sense.

Note that this behavior is also documented [here](https://reactjs.org/docs/refs-and-the-dom.html#accessing-refs): 

> React will assign the `current` property with the DOM element when the component mounts, and assign it back to `null` when it unmounts. ref updates happen before `componentDidMount` or `componentDidUpdate` lifecycle methods.

Reply:
Yes, totally makes sense 🤦🏻 – although it still feels like a workaround to me, happy to change it though.

Reply:
IMHO it is as much workaround as the `key={ siteId }` that we often do to reset the component when the site is switched. Not having strong feelings of course.

Reply:
Fixed via https://github.com/Automattic/wp-calypso/pull/59940/commits/0f1665cfcfbe4779f460fef1ff0e2e22587d913a

Reply:
> IMHO it is as much workaround as the `key={ siteId }` that we often do to reset the component when the site is switched. Not having strong feelings of course.

totally agree 👍🏻